clients/gtest/gemm_gtest.cpp

```diff
@@ -3,7135 +3,135 @@
-gemm_tuple conv_ctest_bwddata_fp32_046 {{1156, 27, 1, 1156, 27, 1156}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_047 {{11664, 147, 1, 11664, 147, 11664}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_048 {{11664, 1600, 1, 11664, 1600, 11664}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_049 {{11664, 363, 1, 11664, 363, 11664}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_050 {{11664, 576, 1, 11664, 576, 11664}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_051 {{11881, 147, 1, 11881, 147, 11881}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_052 {{11881, 363, 1, 11881, 363, 11881}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_053 {{11881, 75, 1, 11881, 75, 11881}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_054 {{12100, 147, 1, 12100, 147, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_055 {{12100, 1600, 1, 12100, 1600, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_056 {{12100, 27, 1, 12100, 27, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_057 {{12100, 363, 1, 12100, 363, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_058 {{12100, 576, 1, 12100, 576, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_059 {{12100, 75, 1, 12100, 75, 12100}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_060 {{121, 1024, 1, 121, 1024, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_061 {{121, 1056, 1, 121, 1056, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_062 {{121, 192, 1, 121, 192, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_063 {{121, 2304, 1, 121, 2304, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_064 {{121, 3456, 1, 121, 3456, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_065 {{121, 363, 1, 121, 363, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_066 {{121, 4, 1, 121, 4, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_067 {{121, 512, 1, 121, 512, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_068 {{121, 75, 1, 121, 75, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_069 {{121, 832, 1, 121, 832, 121}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_070 {{12321, 147, 1, 12321, 147, 12321}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_071 {{12321, 27, 1, 12321, 27, 12321}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_072 {{12321, 363, 1, 12321, 363, 12321}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_073 {{12321, 75, 1, 12321, 75, 12321}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_074 {{12544, 147, 1, 12544, 147, 12544}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_075 {{12544, 1600, 1, 12544, 1600, 12544}, {1, 0}, {'N', 'T'}};
-gemm_tuple conv_ctest_bwddata_fp32_076 {{12544, 27, 1, 12544, 27, 12544}, {1, 0}, {'N', 'T'}};

```

Comment:
Each category (quick, pre_checkin, nightly, known_bug) is instantiated separately, with a filter to select the current category, so that the test name prefix contains the category name.